### ISSUE TITLE
MultiKeys Issue(Found a workaround) and a Tip for RM2 Disconnections

### ISSUE THREAD

**Issue by bdaws** [beckyricha/Broadlink-RM-LAN-Control#22]
I'm new to GitHub but have used your repository extensively over the last few weeks.  First I want to say I appreciate the work you put into this Becky as I would have been lost without your guidance.

Everything was straight forward except I could not get the current Multikeys code to function (Smartthings reports an error in the code but removing the error on that line allows you to add it).  However I could not get it to work as no signals would pass through to the RM2.  I did use your previous revision and have not had any issues (I think it had the line 17 error as well though).

A tip I wanted to add for something I have struggled with regarding the RM2.  It seems that after 4-5 hours of no usage my RM2 would stop working.  It exhibits very strange behavior.  I have a full Meraki setup so I did a lot of going through logs and packet sniffing  to figure out what was going on.  In the logs there would be next to nothing and it actually will show connected and I can ping it from the Meraki cloud.  However on my local network I cannot ping the RM2 when this happens and it will not function unless I restart the AP or the RM2 itself.  Then it's right back to working.  I wondered if others encountered this with the RM2 or if maybe I have a faulty one.

I was discussing this with one of my employees and he had a great idea after we talked for a few minutes that has seemed to solve the disconnection issue.  What I did was setup a dummy code in RM Bridge and then used the automation function in Smartthings to send this dummy code every 4 hours (I have 6 automatons setup with the dummy code occurring every 4 hours).  I first tried 6 hours but ran into the same issue however dropping it to 4 it has been rock solid stable for a few days now.  Hope this helps and once again I appreciate all your work Becky!

**Comment by beckyricha** [comment]
Thanks for the tip. I have never encountered the issue you described with some sort of timeout but this may help anyone else who does. Just wondering though, is your broadlink hub firmware up to date?
On the multi keys no one had reported a bug previously so I'll be happy to fix it if you would please elaborate. Which version is it in?

On Feb 16, 2017 11:54 AM, bdaws <notifications@github.com> wrote:I'm new to GitHub but have used your repository extensively over the last few weeks.  First I want to say I appreciate the work you put into this Becky as I would have been lost without your guidance.
Everything was straight forward except I could not get the current Multikeys code to function (Smartthings reports an error in the code but removing the error on that line allows you to add it).  However I could not get it to work as no signals would pass through to the RM2.  I did use your previous revision and have not had any issues (I think it had the line 17 error as well though).
A tip I wanted to add for something I have struggled with regarding the RM2.  It seems that after 4-5 hours of no usage my RM2 would stop working.  It exhibits very strange behavior.  I have a full Meraki setup so I did a lot of going through logs and packet sniffing  to figure out what was going on.  In the logs there would be next to nothing and it actually will show connected and I can ping it from the Meraki cloud.  However on my local network I cannot ping the RM2 when this happens and it will not function unless I restart the AP or the RM2 itself.  Then it's right back to working.  I wondered if others encountered this with the RM2 or if maybe I have a faulty one.
I was discussing this with one of my employees and he had a great idea after we talked for a few minutes that has seemed to solve the disconnection issue.  What I did was setup a dummy code in RM Bridge and then used the automation function in Smartthings to send this dummy code every 4 hours (I have 6 automatons setup with the dummy code occurring every 4 hours).  I first tried 6 hours but ran into the same issue however dropping it to 4 it has been rock solid stable for a few days now.  Hope this helps and once again I appreciate all your work Becky!

**Comment by bdaws** [comment]
I do have the latest firmware and contacted Broadlink but they were no help.  It's almost as if it goes into a power saving mode or something of that nature.  I saw on some Amazon reviews where other people reported similar issues of it dropping and I have tried multiple routers and had the same issue with each.  I'm on firmware v20025 which is the latest I though.

When I posted the Multikeys code in it said Error on Line 17 I believe and would not let me save it in Smartthings.  However I just tried it again and it didn't give me the error.  Maybe I copied/pasted something incorrectly.  Sorry about that!  I'll try the latest version of the code when I get home.  Thanks!
